src/services/http-req/dogBreedRequest.ts

```diff
@@ -0,0 +1,24 @@
+import fetch from 'node-fetch'
+import AbortController from 'abort-controller'
+import { RequestedDogBreeds } from '../../lambdas/types'
+
+export async function getDogBreeds(): Promise<RequestedDogBreeds> {
+  const controller = new AbortController()
+  let reqTimedOut = false
+  const timeout = setTimeout(() => {
+    controller.abort()
+    reqTimedOut = true
+  }, 1500)
+
+  const res = await fetch('https://dog.ceo/api/breeds/list/all', { signal: controller.signal })
+  const payload = await res.json()
+  clearTimeout(timeout)
+
+  if (reqTimedOut) {
+    const error = new Error('Request to externial API timed out')
+    error.name = 'RequestTimeOut'
+    throw error
+  }
```

Comment:
Your abort controller implementation looks right to me, I would check my comments on your test suite for more info. You're definitely on the right track! If we can remove the need for `reqTimedOut` I think this would be solid.

Reply:
haha yeah the comments in the test suite cleared this up for me, thanks